### PR TITLE
Add skin-controlled cursor rotation

### LINKED ISSUE #6776: Visual - Cursor rotate option
**Describe the new feature:** 
On osu!stable the cursor rotates arround it's centre.
**Proposal designs of the feature:** 
It makes the cursor look less boring in low star maps and when the cursor is textured.

### PR DESCRIPTION
**Pull request by mcendu**
Resolves #6776 

Problems left to address:
- [x] Compare spin rate against stable.

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Skinning/LegacyCursor.cs [hunk 1]
```diff
@@ -30,13 +32,14 @@ private void load(ISkinSource skin)
...
                 },
-                new NonPlayfieldSprite
+                cursor = new NonPlayfieldSprite
                 {
                     Texture = skin.GetTexture("cursor"),
                     Anchor = Anchor.Centre,
                     Origin = Anchor.Centre,
                 }
             };
+            cursor.Spin(10000, RotationDirection.Clockwise);
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Skinning/LegacyCursor.cs]: This duration should probably be compared with stable's - did you check if the rate of spins is the same?

**Comment by bdach** [comment]
What do you mean by "not initiating rotation all the time"? Did you try to debug that?

File: osu.Game.Rulesets.Osu/Skinning/LegacyCursor.cs [hunk 2]
```diff
@@ -11,14 +11,19 @@ namespace osu.Game.Rulesets.Osu.Skinning
...
             Size = new Vector2(50);
 
             Anchor = Anchor.Centre;
             Origin = Anchor.Centre;
+
+            rotate = spin;
         }
 
+        private NonPlayfieldSprite cursor;
+        private readonly bool rotate;
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/Skinning/LegacyCursor.cs]: Rename to `spin`.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by smoogipoo** [comment]
Thanks. I've moved the retrieval into LegacyCursor to follow the existing pattern inside `LegacySliderBall`.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**